### DISCUSSION TITLE
Custom Ruleset Contest idea

### DISCUSSION THREAD

**Discussion by Tenexxt** [ppy/osu#28953] [Ideas] [answered]
I love lazer a lot, it broght me back to osu after a **VERY** long break, the ui, the everything - i love it. Tho i see that a lot of people don't share the same feelings for the Beta of the next major update.
Many complain about the editor, the skining for lazer, and the list sadly goes on. Tho i don't complain about those (mostly), i love the argon skin, i love the ui, sfx - **but there is one thing the game is lacking to me:**

Orginal features

But not in the sense "The game isn't orginal.", but in a sense **"We need more Lazer features only."**!

We are getting very soon the "Daily Challenge" for lazer, which goal is: Bringing more people to lazer, and expanding osu as a whole.
In all aspect of osu there can be competition: Fanart, Mapping, Skining, and of course Gameplay. But i dont see one aspect of Lazer, or osu as a whole talked a lot: **Developer**. And no, **not the development of osu**, but of **custom rulesets**. 

In 2007, almost 20 years ago we got the first version of osu!, in 2008 we got osu!taiko, in 2010 we got osu!catch, and lastly in 2012 we got osu!mania. **It just ends there.**

osu!lazer is all about accessibility - we got a version for phones, way more mods, and soon we are getting the DC, and there are more thing which i didn't list. What i didn't list is custom rulesets - why? because they aren't talked about much, and they open a lot of possibilities for osu, and that makes me sad. So many good ideas, and most don't know about them. 

**Thats why i thought of CRC** - it brings new features for lazer, can expand the community a lot, and will make Lazer even better as a major update - because people wont be able to say 'not much was added'

**CRC (Full name: Custom Ruleset Contest**) is a contest that could get hosted once, or once a few years.
The idea is simple - **The winning ruleset is going to be worked on with main osu devs to be officaly added to the game, and be the next main osu! gamemode.**
All main osu! modes were inspired by something, and CRC will introduce **a lot of new orginal osu ideas**, inspired ones and Custom Rulesets that already exist to a lot of people, and **this form of competition could bring more amazing ideas from the community and works from coders**. This would really show to people, that osu isn't ending at where it is - Its expanding, and growing as a community.

There would be two types of submissions: **Ideas and Projects.**

- Ideas: Submissions to vote for filtered by the NAT, so that random content, or ones extremely vauge isn't votable. Ideas have the same value as Projects, **but are supposed to be made mostly of orginal ideas for gamemodes**
- Projects: Custom Rulesets that already exist, and its devs decided that their idea is in somewhat promotable state. **Somewhat promotable aka playable** - it can have bugs, optimisation issues, **but has to show the gameplay atleast, if it doesn't - submit it as an idea**

_**Note: Of course, the CRs dont have to be polished to be submitted. After all, their devs will work with the osu!dev team to bring their idea to a state it fits lazer and is atleast ready for to be relased officaly, and later polished**_

**The voting would have a point system:**

- NAT and GMT: **1000 points** - People who work for the development of osu deserve to have a higher say, since they will have to help into making the winning idea. this also aplies to peppy btw
- Alumni and BN: **10 points** - People who help push the game forward already, they have a bit more to say than regular players
- Normal players - **1 point.**

**If someone has multiple ranks, the highest point one is automatically selected.**

**Voting requirements**: 24h of minimum playtime, 500pp on any mode - Reason? so the vote will be harder to vote, Account older than 14 days
**Voting rules**: every player can vote for 5 submissions in total.

**Submission requriements:** 
- Ideas: Orginal gamemode ideas, with a throught explenation on how they work (objects, Stats, Ranking system, and all similar details and other important stuff)
- Projects: Minimum state - Wip/Playable - enough to give the whole concept on how the mode works. Its encouraged to be a orginal idea, but anything is good enough. Made using C# code language (C# - CSharp, for people who dont know)

**Timetable**

- Submission time - 2 weeks. You can't make a very polished project by yourself in that time, but you have ideas for that. Dont rush to make something that will break the game when you open it, focus on polishing the concept, and submit it.
- Judging time - 2 weeks. People who would've voted would, and this 14 day limit could limit the amount of bots.
- Winner anouncements - ASAP. Since it has a point system, all you have to do is form a site for it and get the correct data.
- Making time - Max 3 years. It doesn't have to be extemely polished on relase, but also needs to work. the most important part is so that the gameplay and the editor will work, but there is a time limit, so that you, the devs wont hold it off to the last minute, and so that the community doesn't forget about it.

I think thats it, if you have any questions, ask in the comments.
**Thanks for reading about the Custom Ruleset Contest**

Edit: Made the point system more balanced

**Comment by peppy** [comment, marked as answer]
I know you wrote a lot and haven't got a response, so sorry for that. You sound very excited to make this happen, but I think we just don't have the spare manpower to do so right now. We're struggling with the 4 rulesets we current have and really need a year or two before we'd be willing to "rank" any new ones.

But things may change if someone comes up with a really cool rulese[t](https://github.com/peppy/ruleset-typer) that changes everything, we'll see.
